meilisearch-http/src/index_controller/dump_actor/loaders/v1.rs

```diff
@@ -60,20 +60,13 @@ impl MetadataV1 {
 #[derive(Default, Clone, Serialize, Deserialize, Debug)]
 #[serde(rename_all = "camelCase", deny_unknown_fields)]
 struct Settings {
-    #[serde(default, deserialize_with = "deserialize_some")]
-    pub ranking_rules: Option<Option<Vec<String>>>,
-    #[serde(default, deserialize_with = "deserialize_some")]
-    pub distinct_attribute: Option<Option<String>>,
-    #[serde(default, deserialize_with = "deserialize_some")]
-    pub searchable_attributes: Option<Option<Vec<String>>>,
-    #[serde(default, deserialize_with = "deserialize_some")]
-    pub displayed_attributes: Option<Option<BTreeSet<String>>>,
-    #[serde(default, deserialize_with = "deserialize_some")]
-    pub stop_words: Option<Option<BTreeSet<String>>>,
-    #[serde(default, deserialize_with = "deserialize_some")]
-    pub synonyms: Option<Option<BTreeMap<String, Vec<String>>>>,
-    #[serde(default, deserialize_with = "deserialize_some")]
-    pub attributes_for_faceting: Option<Option<Vec<String>>>,
+    pub ranking_rules: Setting<Vec<String>>,
+    pub distinct_attribute: Setting<String>,
+    pub searchable_attributes: Setting<Vec<String>>,
+    pub displayed_attributes: Setting<BTreeSet<String>>,
+    pub stop_words: Setting<BTreeSet<String>>,
+    pub synonyms: Setting<BTreeMap<String, Vec<String>>>,
+    pub attributes_for_faceting: Setting<Vec<String>>,
```

Comment:
If we change these settings back to the `Option<Option<T>>` type it should be ok.

Reply:
I am in fact asking the help of @MarinPostma who worked on the dumps, he will be able to tell us if it is safe to change these struct fields types.

Reply:
After discussing with @irevoire and @MarinPostma we think it would be better to keep the legacy settings types even if I tested it and it works great. Thank you for your time :)

Reply:
yep, agree. Reverted to `Option<Option<T>>` and added conversion to `Setting<T>`